src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs

```diff
@@ -455,38 +497,45 @@ private void UpdateActiveProfile()
         public async virtual System.Threading.Tasks.Task SaveLaunchSettings()
         {
             ILaunchSettingsProvider provider = GetDebugProfileProvider();
-            if (EnvironmentVariables != null && EnvironmentVariables.Count > 0)
+            if (EnvironmentVariables != null && EnvironmentVariables.Count > 0 && SelectedDebugProfile != null)
             {
-                SelectedDebugProfile.MutableEnvironmentVariables = EnvironmentVariables.CreateDictionary();
+                SelectedDebugProfile.EnvironmentVariables.Clear();
+                foreach(var kvp in EnvironmentVariables)
+                {
+                    SelectedDebugProfile.EnvironmentVariables.Add(kvp.Name, kvp.Value);
```

Comment:
Is it possible for the SelectedDebugProfile to change while this async task is running?